src/parquet/column/reader.h

```diff
@@ -184,9 +188,9 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
-    if (num_def_levels != num_rep_levels) {
+    if (def_levels && num_def_levels != num_rep_levels) {
```

Comment:
I believe edani's purpose of the `def_levels`/ `rep_levels` check in the `if condition` is to provide users a way to skip reading these values altogether. A user can choose not to read these values inspite of `max_definition_level > 0` and `max_repetition_level > 0` by passing NULL.